ddtrace/tracer/tracer_test.go

```diff
@@ -922,6 +922,28 @@ func TestTracerBaggageImmutability(t *testing.T) {
 	assert.Equal("changed!", childContext.baggage["key"])
 }
 
+func TestTracerInjectConcurrency(_ *testing.T) {
+	tr := newTracer()
+	defer tr.Stop()
+	span, _ := StartSpanFromContext(context.Background(), "main")
+	defer span.Finish()
+
+	var wg sync.WaitGroup
+	for i := 0; i < 500; i++ {
+		wg.Add(1)
+		i := i
+		go func(val int) {
+			defer wg.Done()
+			span.SetBaggageItem("val", fmt.Sprintf("%d", val))
+
+			traceContext := map[string]string{}
+			_ = tr.Inject(span.Context(), TextMapCarrier(traceContext))
+		}(i)
+	}
+
+	wg.Wait()
+}
+
```

Comment:
@darccio when I run this test against main I can't get the test to fail with `-race` enabled, no matter how many runs I give it. It looks like `newTracer()` doesn't set up the tracer in the right way to trigger this. I will make a commit with the necessary changes that make this fail consistently for me locally.

The other test, `TestConcurrentSetAttributes`, fails easily for me.